### PR TITLE
feat(aio): add footer background

### PR DESCRIPTION
**Pull request by sjtrimble**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
Removed duplicate media query code (also in mixin)

### CODE REVIEW

File: aio/src/app/app.component.html [hunk 1]
```diff
@@ -24,4 +24,6 @@
 
 <aio-search-results #searchResults></aio-search-results>
 
-<aio-footer [nodes]="footerNodes" [versionInfo]="versionInfo" ></aio-footer>
+<footer>
```
Comment (petebacondarwin) [inline aio/src/app/app.component.html]: Why move the footer element here from the aio-footer?

Comment (sjtrimble) [inline aio/src/app/app.component.html]: The ::after wasn't quite working on the component, so I switched it while trying to figure it out, and then I  decided it might make more sense semantically anyway (to have the component inside of the footer). Is this ok or did I break a major Angular rule 😱

Comment (petebacondarwin) [inline aio/src/app/app.component.html]: Not a big deal just looked like an unnecessary change. Not a blocker :-)

File: aio/src/app/app.component.html [hunk 2]
```diff
@@ -3,7 +3,7 @@
     (click)="sidenav.toggle()" title="Docs menu">
     <md-icon>menu</md-icon>
   </button>
-  <a class="nav-link home" href="/"><img src="{{ homeImageUrl }}" title="Home" alt="Home"></a>
+  <a class="nav-link home" href="/"><img class="home-logo-link" src="{{ homeImageUrl }}" title="Home" alt="Home"></a>
```
Comment (petebacondarwin) [inline aio/src/app/app.component.html]: why this additional class when you already have `.nav-link.home > img`?

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
I am afraid that the AppComponent unit tests are broken by the change to the footer.

https://travis-ci.org/angular/angular/jobs/217036824#L1173

This test is failing: https://github.com/angular/angular/blob/master/aio/src/app/app.component.spec.ts#L172-L177

```

  describe('footer', () => {
    it('should have version number', () => {
      const versionEl: HTMLElement = fixture.debugElement.query(By.css('aio-footer p')).nativeElement;
      expect(versionEl.innerText).toContain(TestHttp.versionFull);
    });
  });
```

Arguably that test is not valid in terms of a unit test because the `AppComponent` shouldn't be aware of how the `FooterComponent` is rendering itself.

The simple fix is to remove the `p` from the selector, I believe.

**Review by petebacondarwin — APPROVED** [review]

**Comment by wardbell** [comment]
Yes .. `<aio-footer>` needs its own tests. We're going to have to go back and fill in gaps. No time to do that before next week. But during next week would be wise.
